ui/app/src/components/RemoteRepositoriesStatus/RemoteRepositoriesStatus.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2007-2021 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState } from 'react';
+import { Resource } from '../../models/Resource';
+import { RepositoryStatus } from '../../models/Repository';
+import RemoteRepositoriesStatusUI from './RemoteRepositoriesStatusUI';
+import CommitResolutionDialog from '../CommitResolutionDialog/CommitResolutionDialog';
+import { cancelFailedPull, resolveConflict as resolveConflictService } from '../../services/repositories';
+import { useActiveSiteId } from '../../utils/hooks';
+import { useDispatch } from 'react-redux';
+import { showSystemNotification } from '../../state/actions/system';
+import { showErrorDialog } from '../../state/reducers/dialogs/error';
+import { defineMessages, useIntl } from 'react-intl';
+import RemoteRepositoriesDiffDialog from '../RemoteRepositoriesDiffDialog';
+
+const messages = defineMessages({
+  revertPullSuccessMessage: {
+    id: 'repositories.revertPullSuccessMessage',
+    defaultMessage: 'Successfully reverted repository.'
+  },
+  commitSuccessMessage: {
+    id: 'repositories.commitSuccessMessage',
+    defaultMessage: 'Successfully committed.'
+  }
+});
+
+export interface RemoteRepositoriesStatusProps {
+  resource: Resource<RepositoryStatus>;
+  setFetching(fetching): void;
+  onActionSuccess?(status): void;
+}
+
+export default function RemoteRepositoriesStatus(props: RemoteRepositoriesStatusProps) {
```

Comment:
Not clear what this is from the name

Reply:
This is actually studio's repository status rather than the remotes. Rename to `StudioRepositoryStatus` (including all of it's subcomponents)